client/client.go

```diff
@@ -47,13 +57,47 @@ func (c Client) Logger() hclog.Logger {
 
 // withProject allows multiplexer to create a new client with given subscriptionId
 func (c Client) withProject(project string) *Client {
-	return &Client{
-		backoff:   c.backoff,
-		projects:  c.projects,
-		Services:  c.Services,
-		logger:    c.logger.With("project_id", project),
-		ProjectId: project,
+	c.logger = c.logger.With("project_id", project)
+	c.ProjectId = project
+	return &c
+}
+
+func (c *Client) initServiceusageData() {
+	for _, p := range c.projects {
+		cl := c.withProject(p)
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+		defer cancel()
+		var err error
+		c.ServiceusageData[p], err = cl.fetchServiceusageServices(ctx)
+		if err != nil {
+			cl.Logger().Warn("failed to get a list of disabled APIs", "error", err, "project", p)
+			continue
+		}
+		for _, s := range c.ServiceusageData[p] {
+			if s.State == "DISABLED" {
+				c.DisabledServices[p][s.Config.Name] = struct{}{}
+			}
+		}
+	}
+}
+
+func (c *Client) fetchServiceusageServices(ctx context.Context) ([]*serviceusage.GoogleApiServiceusageV1Service, error) {
+	items := make([]*serviceusage.GoogleApiServiceusageV1Service, 0, 1000)
+	nextPageToken := ""
+	for {
+		call := c.Services.ServiceUsage.Services.List(fmt.Sprintf("projects/%s", c.ProjectId)).PageToken(nextPageToken)
```

Comment:
For the `cq-playground` project there are currently over 2200 services that are returned here. We are getting 50 services at time so it takes about 44 calls to fully resolve. I suggest that we make 2 changes to ensure 
we are grabbing the minimal amount of data during the initialization phase:
1. Change from 50 items/ per request to 100
2. Only fetch the enabled services.. This will drastically reduce the total number of services that are grabbed. It will mean that the disabled services will have to be fetched in the resource resolver.
```suggestion
		call := c.Services.ServiceUsage.Services.List(fmt.Sprintf("projects/%s", c.ProjectId)).Filter(`state:ENABLED`).PageSize(100).PageToken(nextPageToken)
```